portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,8 +39,6 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         PrintWriter writer = response.getWriter();
 
         response.setContentType("text/html;");
-        
-        writer.println("<h1>Hello world!</h1>");
-        writer.println("<h2>Greeting from " + renderedName + "!</h2>");
```

Comment:
Prefer
 String.format("<h2>Greeting from %s!</h2>", renderedName)

Reply:
Does this also avoid any injection stuff?

Reply:
Nope. It's not content aware, and doesn't pursue such goal (One might want to use String.format to insert html snippets into a string, "html" is just one possible example it basically can be anything)